contracts/interfaces/IDynamicFeeManager.sol

```diff
@@ -7,7 +7,5 @@ import {IPoolManager} from "./IPoolManager.sol";
 /// @notice The dynamic fee manager determines fees for pools
 /// @dev note that this pool is only called if the PoolKey fee value is equal to the DYNAMIC_FEE magic value
 interface IDynamicFeeManager {
-    function getFee(address sender, PoolKey calldata key, IPoolManager.SwapParams calldata params, bytes calldata data)
-        external
-        returns (uint24);
+    function getFee(address sender, PoolKey calldata key) external view returns (uint24);
```

Comment:
params seem pretty important here (hooks may change fee based on size or direction of swap). Can't find gas benchmark notion, but guessing the benchmarks assumed beforeSwap gets called and sets the dynamic fee on every swap?? (likely tstore-ing some information about the swap in the hook for the call)

Reply:
agree and thought that the ability to pass through SwapParams to the hook would be a compelling use case for dynamic fees

Reply:
If you want to change the fee based on the swap you already have them passed through in`beforeSwap`. If you look at the test [here](https://github.com/Uniswap/v4-core/blob/dynamic-fee/contracts/test/DynamicFeesTest.sol#L28) you can see that the DynamicFee hook has all the info it needs already. The `getFee` function is just the call to the contract made in `_fetchDynamicFee` so there are not even params to pass through.

Reply:
What's the purpose of adding "view" for `getFee`. I guess there would be use cases that dynamic update fee state in getFee.

Reply:
Is there a reason to make it not view?